libraries/AP_Compass/AP_Compass_IST8308.cpp

```diff
@@ -0,0 +1,246 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
```

Comment:
spdx is just a simplification... I can update it to the full license, but we have so many examples of files in the codebase **without** any license that I don't feel the need for it.

Regarding GPL2. This is 2-**or-later** because it allows anyone besides me to base on this implementation to add it to the Linux kernel.  I remember a conversation a long-time ago with @tridge about it and it was deemed ok.

Reply:
> spdx is just a simplification... I can update it to the full license, but we have so many examples of files in the codebase without any license that I don't feel the need for it.

In my opinion they should be in all files, but what I'd like to avoid is making it worse by having yet another different way.

> Regarding GPL2. This is 2-or-later because it allows anyone besides me to base on this implementation to add it to the Linux kernel. I remember a conversation a long-time ago with @tridge about it and it was deemed ok.

OK, that's not my understanding. If that's the header we have there, it is saying this file is distributed **by us** under GPLv2. It is saying that any modification that happens to this file is under the GPLv2.

If you want to distribute this code under GPLv2 or later, that's fine, but I don't think ArduPilot should distribute it under the GPLv2 license.

I'm not completely sure if with the GPLv2 or later statement we have to mandatory include it even if we re-license it as GPLv3 or later, but I think that would be fine - we have some code from other projects and their license is obviously there.

Reply:
I went back to full blown version instead of the shorter spdx. I left the 

> it is saying this file is distributed **by us** under GPLv2

no, it's v2 **or any later version**. There's a huge difference - it basically allows to be compatible with v2 while still allowing the whole to be v3, which is what really matter for us.

I left this as is for the next version